clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaAdminClientConfigProperties.java

```diff
@@ -60,9 +60,13 @@ public final void setAdminClientConfig(final Map<String, String> adminConfig) {
 
     /**
      * Gets the Kafka admin client configuration. This is the result of applying the admin client configuration on the
-     * common configuration. The returned map will contain a property {@code client.id} which will be set to a unique
-     * value containing the already set client id or alternatively the value set via
-     * {@link #setDefaultClientIdPrefix(String)}, the given adminClientName and a newly created UUID.
+     * common configuration.
+     * <p>
+     * It is ensured that the returned map contains a unique {@code client.id}. The client ID will be created from the
+     * given client name, followed by a component identifier with included or added unique ID (either containing pod name
+     * and container ID if running in Kubernetes, or including component unique ID or name set via
+     * {@link #setComponentUid(String)}) or {@link #setComponentName(String)})).
```

Comment:
That's hard to understand in prose. Perhaps examples would help, or a schematic diagram listing the individual elements and their manifestations. 
Alternatively, one could perhaps refrain from explaining the whole algorithm in detail in the JavaDoc. If it is complex and you can never be sure if the documentation is up to date and covers all edge cases anyway, I might rather look directly into the implementation than work through the text first, then look at the implementation and then think about possible differences...

Reply:
I would opt for the alternative then: I think there's no need to explain the algorithm in detail here - it should be enough to document the fact that a unique id is used as a suffix.
I've therefore removed the part in parentheses above.